crates/router/src/services/authentication.rs

```diff
@@ -101,6 +102,32 @@ impl AuthenticateAndFetch<storage::MerchantAccount> for PublishableKeyAuth {
     }
 }
 
+#[derive(Debug)]
+pub struct JWTAuth;
+
+#[derive(serde::Deserialize)]
+struct JwtAuthPayloadFetchMerchantAccount {
+    merchant_id: String,
+}
+
+#[async_trait]
+impl AuthenticateAndFetch<storage::MerchantAccount> for JWTAuth {
+    async fn authenticate_and_fetch(
+        &self,
+        request_headers: &HeaderMap,
+        state: &AppState,
+    ) -> RouterResult<storage::MerchantAccount> {
+        let mut token = get_jwt(request_headers)?;
+        token = strip_jwt_token(token)?;
+        let payload = decode_jwt::<JwtAuthPayloadFetchMerchantAccount>(token, state)?;
+        state
+            .store
+            .find_merchant_account_by_merchant_id(&payload.merchant_id)
```

Comment:
Just a question, if a DB call has to happen for authentication then what is the advantage of using JWT token ? Wouldn't  a similar auth like an api key suffice ?

Reply:
- JWT can contain more information in payload and not strictly be attached to getting merchant account. This is just one of the implementation. Whereas, API key in our case is strictly attached to a merchant account.
- API key is generally used for S2S calls, whereas JWT will be used for client to server call, where token will be issued for every session and can be expired (i guess API keys cant be expired based on time).